### PR TITLE
Bug 1449908 - Group replica sets by owner reference

### PR DESCRIPTION
**Pull request by spadgett**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1449908

Use owner references to decide if a replica set is owned by a deployment. Change maps to use deployment UID instead of deployment name. This avoids problems when a deployment is deleted and another is created with the same name.

openshift web console 2017-05-17 12-09-50

### CODE REVIEW

File: app/scripts/controllers/newOverview.js [hunk 1]
```diff
@@ -1195,13 +1197,13 @@ function OverviewController($scope,
       group: "extensions",
       resource: "deployments"
     }, context, function(deploymentData) {
-      overview.deployments = deploymentData.by('metadata.name');
+      overview.deploymentsByUID = deploymentData.by('metadata.uid');
```
Comment (jwforres) [inline app/scripts/controllers/newOverview.js]: this isnt new, but were we not sorting deployments before?  how is it being ordered when you are looking at the by Resource Type view?

File: app/scripts/services/ownerReferences.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  .factory("OwnerReferencesService", function() {
+    var getOwnerReferences = function(apiObject) {
+      return _.get(apiObject, 'metadata.ownerReferences');
+    };
+    return {
+      getOwnerReferences: getOwnerReferences,
+      groupByControllerUID: function(apiObjects) {
+        return _.groupBy(apiObjects, function(apiObject) {
+          var ownerRefs = getOwnerReferences(apiObject);
+          var controllerRef = _.find(ownerRefs, { controller: true });
```
Comment (jwforres) [inline app/scripts/services/ownerReferences.js]: is it possible (in theory) for more than one controller to claim ownership, if that were the case you would want the item to appear under each group and not just the first one you find.   Would be curious if the API prevents you from having more than one ownerRef with controller true.  Could try manually adding an ownerRef on a pod that points to a random controller that exists.

File: app/scripts/controllers/newOverview.js (hunk 1 above)
Comment (spadgett) [inline app/scripts/controllers/newOverview.js]: Looks like an oversight and they weren't sorted.

**Comment by spadgett** [comment]
@jwforres updated

File: app/scripts/controllers/newOverview.js (hunk 1 above)
Comment (jwforres) [inline app/scripts/controllers/newOverview.js]: can you open an issue for this so we dont lose track of it

**Comment by jwforres** [comment]
[merge]

File: app/scripts/controllers/newOverview.js (hunk 1 above)
Comment (spadgett) [inline app/scripts/controllers/newOverview.js]: Opened #1555

**Pull request merged**